cmd/collectors/rest/plugins/certificate/certificate.go

```diff
@@ -0,0 +1,280 @@
+	var (
+		cert *x509.Certificate
+		err  error
+	)
+
+	certificatePEM := instance.GetLabel("certificatePEM")
+	certUuid := instance.GetLabel("uuid")
+
+	if certificatePEM == "" {
+		my.Logger.Warn().Str("uuid", certUuid).Msg("Certificate is not found")
+		instance.SetLabel("certificateIssuerType", "unknown")
+	} else {
+		instance.SetLabel("certificateIssuerType", "self_signed")
+		certDecoded, _ := pem.Decode([]byte(certificatePEM))
+		if certDecoded == nil {
+			my.Logger.Warn().Msg("PEM formatted object is not a X.509 certificate. Only PEM formatted X.509 certificate input is allowed")
+			instance.SetLabel("certificateIssuerType", "unknown")
+			return
+		}
+
+		if cert, err = x509.ParseCertificate(certDecoded.Bytes); err != nil {
+			my.Logger.Warn().Msg("PEM formatted object is not a X.509 certificate. Only PEM formatted X.509 certificate input is allowed")
+			instance.SetLabel("certificateIssuerType", "unknown")
+			return
+		}
+
+		// Verifies if certificate is self-issued. This is true if the subject and issuer are equal.
+		if cert.Subject.String() == cert.Issuer.String() {
+			// Verifies if certificate is self-signed. This is true if the certificate is signed using its own public key.
+			if err = cert.CheckSignature(x509.SHA256WithRSA, cert.RawTBSCertificate, cert.Signature); err != nil {
+				// Any verification exception means it is not signed with the give key. i.e. not self-signed
+				instance.SetLabel("certificateIssuerType", "ca_signed")
+			}
+		}
+	}
+}
+
+func (my *Certificate) setCertificateValidity(data *matrix.Matrix, instance *matrix.Instance) {
+	var (
+		expiryTimeMetric matrix.Metric
+	)
+
+	instance.SetLabel("certificateExpiryStatus", "unknown")
+
+	if expiryTimeMetric = data.GetMetric("expiry_time"); expiryTimeMetric == nil {
+		my.Logger.Error().Stack().Msg("missing expiry time metric")
+		return
+	}
+
+	if expiryTime, ok := expiryTimeMetric.GetValueFloat64(instance); ok {
+		// convert expiryTime from float64 to int64 and find difference
+		timestampDiff := time.Unix(int64(expiryTime), 0).Sub(time.Now()).Hours()
+
+		if timestampDiff <= 0 {
+			instance.SetLabel("certificateExpiryStatus", "expired")
+		} else {
+			// daysRemaining will be more than 0 if it has reached this point, convert to days
+			daysRemaining := timestampDiff / 24
+			if daysRemaining < 60 {
+				instance.SetLabel("certificateExpiryStatus", "expiring")
+			} else {
+				instance.SetLabel("certificateExpiryStatus", "active")
+			}
+		}
+	}
+
+}
+
+func (my *Certificate) setPluginInterval() (int, error) {
+
+	volumeDataInterval := my.getDataInterval(my.ParentParams, DefaultDataPollDuration)
+	pluginDataInterval := my.getDataInterval(my.Params, DefaultPluginDuration)
+	my.Logger.Debug().Float64("VolumeDataInterval", volumeDataInterval).Float64("PluginDataInterval", pluginDataInterval).Msg("Poll interval duration")
+	my.pluginInvocationRate = int(pluginDataInterval / volumeDataInterval)
```

Comment:
is it ok if this is zero? will happen when pluginDataInterval < volumeDataInterval

Reply:
is it ok if this is zero? 
--> it should not zero, as pluginDataInterval would be either given in plugin config or default is ``` DefaultPluginDuration = 30 * time.Minute```
--> If it is zero, that should be fine, just it will invoke the plugin flow(costlier) at every data poll(3min).